### ISSUE TITLE
Character Camera and Flash Light

### ISSUE THREAD

**Issue by Morbott** [TheGameCreators/GameGuruRepo#5148]
I was setting up a scene and two things came up. The camera and the player flash-light. I was wondering if it would be better to have the Camera and Flash light their own separate objects. Then, in their properties you can choose what is their parent, and all their properties, and the ability to adjust their position with the parent. For example, the position of the flash light. Higher... lower... maybe on the left side.. maybe on the right... maybe on the head? 
I find it odd that all the camera properties are hidden and can only be modified while the game is playing. Likewise the player flashlight has no properties at all. It's just a spotlight connected to the character no? So if not making it a separate object that is parented to the player, why not give it all the light properties so one can customize it. While on the topic of flash light, it would be nice to be able to give it a texture so that it looks more like a flash light when on. Also a lag feature would be nice, so when the player moves, there's a bit of lag on the light.
So, in a nutshell:

 Camera
-Separate Object
-Camera settings available outside of the run-time.

Flash light
-Separate Object ( or just make it the spot light with Flash light behaviour )
-access to all flash/spot light parameters (brightness, radius, distance, colour)
-ability to project texture
-lag feature

I guess part of this would mean that there is a parenting system in GGMax. Which might also mean that GGMax would need an outliner.
Thanks
M
